test/plugin_test.py

```diff
@@ -1473,6 +1475,96 @@ def test_invalid_uri(self):
+                "volume cache info query")
+
+    def test_volume_cache_set(self):
+        flag_tested = False
+        for s in self.systems:
+            cap = self.c.capabilities(s)
+            if not supported(cap, [Cap.VOLUME_CACHE_INFO]):
+                continue
+
+            lsm_vol = self._volume_create(s.id)[0]
+            if supported(cap, [Cap.VOLUME_PHYSICAL_DISK_CACHE_SET]):
+                self.c.volume_physical_disk_cache_set(
+                    lsm_vol, lsm.Volume.PHYSICAL_DISK_CACHE_ENABLED)
+                cache_info = self.c.volume_cache_info(lsm_vol)
+                self.assertTrue(cache_info[4] ==
+                                lsm.Volume.PHYSICAL_DISK_CACHE_ENABLED)
+                self.c.volume_physical_disk_cache_set(
+                    lsm_vol, lsm.Volume.PHYSICAL_DISK_CACHE_DISABLED)
+                cache_info = self.c.volume_cache_info(lsm_vol)
+                self.assertTrue(cache_info[4] ==
+                                lsm.Volume.PHYSICAL_DISK_CACHE_DISABLED)
+
+            if supported(cap, [Cap.VOLUME_WRITE_CACHE_POLICY_SET_WRITE_BACK]):
+                self.c.volume_write_cache_policy_set(
+                    lsm_vol, lsm.Volume.WRITE_CACHE_POLICY_WRITE_BACK)
+                cache_info = self.c.volume_cache_info(lsm_vol)
+                self.assertTrue(cache_info[0] ==
+                                lsm.Volume.WRITE_CACHE_POLICY_WRITE_BACK)
+
+            if supported(cap, [Cap.VOLUME_WRITE_CACHE_POLICY_SET_AUTO]):
+                self.c.volume_write_cache_policy_set(
+                    lsm_vol, lsm.Volume.WRITE_CACHE_POLICY_AUTO)
+                cache_info = self.c.volume_cache_info(lsm_vol)
+                self.assertTrue(cache_info[0] ==
+                                lsm.Volume.WRITE_CACHE_POLICY_AUTO)
+
+            if supported(cap,
+                         [Cap.VOLUME_WRITE_CACHE_POLICY_SET_WRITE_THROUGH]):
+                self.c.volume_write_cache_policy_set(
+                    lsm_vol, lsm.Volume.WRITE_CACHE_POLICY_WRITE_THROUGH)
+                cache_info = self.c.volume_cache_info(lsm_vol)
+                self.assertTrue(cache_info[0] ==
+                                lsm.Volume.WRITE_CACHE_POLICY_WRITE_THROUGH)
+
+            if supported(cap, [Cap.VOLUME_READ_CACHE_POLICY_SET]):
+                self.c.volume_read_cache_policy_set(
+                    lsm_vol, lsm.Volume.READ_CACHE_POLICY_ENABLED)
+                cache_info = self.c.volume_cache_info(lsm_vol)
+                self.assertTrue(cache_info[2] ==
+                                lsm.Volume.READ_CACHE_POLICY_ENABLED)
+                self.c.volume_read_cache_policy_set(
+                    lsm_vol, lsm.Volume.READ_CACHE_POLICY_DISABLED)
+                cache_info = self.c.volume_cache_info(lsm_vol)
+                self.assertTrue(cache_info[2] ==
+                                lsm.Volume.READ_CACHE_POLICY_DISABLED)
+
+            self._volume_delete(lsm_vol)
+            flag_tested = True
```

Comment:
This flag only really proves if the hardware support VOLUME_CACHE_INFO as all the other tests could fail to even test if not supported.